x-pack/platform/plugins/shared/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.tsx

```diff
@@ -378,7 +378,8 @@ export const ShareToSpaceFlyoutInternal = (props: ShareToSpaceFlyoutProps) => {
         enableCreateCopyCallout &&
         spaces.length > 1 &&
         savedObjectTarget.namespaces.length === 1 &&
-        !arraysAreEqual(savedObjectTarget.namespaces, [ALL_SPACES_ID]);
+        !arraysAreEqual(savedObjectTarget.namespaces, [ALL_SPACES_ID]) &&
+        !savedObjectTarget.type.startsWith('ml');
```

Comment:
Nit: The only concern I have here is that we're string matching to catch all ML SO types. This appears to work, but I am unfamiliar with all ML SO types. Could we make this more explicit, something like an "isMLType" utility function that we can ensure covers only, but all of, the ML SO types explicitly?